src/main/java/org/prebid/server/bidder/sharethrough/SharethroughMarkupUtil.java

```
@@ -0,0 +1,64 @@
+                .append("data-stx-response-name=\"").append(strRespId).append("\"></div>\n")
+                .append("\t\t<script>var ").append(strRespId).append(" = ")
+                .append("\"").append(encodedJsonResponse).append("\"</script>\n");
+
+        if (BooleanUtils.toBoolean(strUriParameters.getIframe())) {
+            tmplBody.append("\n\t\t<script src=\"//native.sharethrough.com/assets/sfp.js\"></script>\n");
+        } else {
+            tmplBody.append("\t\t\t<script src=\"//native.sharethrough.com/assets/sfp-set-targeting.js\"></script>\n")
+                    .append("\t    \t<script>\n")
+                    .append("\t     (function() {\n")
+                    .append("\t     if (!(window.STR && window.STR.Tag) && !(window.top.STR && window.top.STR.Tag)){\n")
+                    .append("\t         var sfp_js = document.createElement('script');\n")
+                    .append("\t         sfp_js.src = \"//native.sharethrough.com/assets/sfp.js\";\n")
+                    .append("\t         sfp_js.type = 'text/javascript';\n")
+                    .append("\t         sfp_js.charset = 'utf-8';\n")
+                    .append("\t         try {\n")
+                    .append("\t             window.top.document.getElementsByTagName('body')[0].appendChild(sfp_js);\n")
+                    .append("\t         } catch (e) {\n")
+                    .append("\t           console.log(e);\n")
+                    .append("\t         }\n")
+                    .append("\t       }\n")
+                    .append("\t     })()\n").append("\t\t   </script>\n");
+        }
+        return tmplBody.toString();
+    }
+
+    private static String parseResponseToString(ExtImpSharethroughResponse response) {
+        try {
+            return Json.mapper.writeValueAsString(response);
+        } catch (JsonProcessingException e) {
+            return null;
```

You should throw an actual exception here and you won't need to do additional checks in `getAdMarkup` or `toBidderBid`. In Go it doesn't matter what they return because whenever an error occurs they will return it right away.

You just add yourself unnecessary checks - just throw an exception and catch it up in `makeBids`.